.github/workflows/dart-tests.yaml

```diff
@@ -1,135 +1,104 @@
-name: Relic CI
+name: Relic Ci
 
 on:
   push:
-    branches:
-      - main
-      - dev
+    branches: [main]
   pull_request:
-    branches:
-      - main
-      - dev
+    branches: [main]
+  schedule:
+    - cron: "0 0 * * 0" # every Sunday at 00:00 UTC
+  workflow_dispatch: # allow manual triggering
 
 env:
   PUB_CACHE_PATH: ~/.pub-cache
 
 jobs:
-  dart_format:
-    name: Verify Dart Formatting
+  build:
+    name: Verify Build
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [windows-latest, ubuntu-latest, macos-latest]
-        dart_sdk: ['3.3.0']
+        os: [ubuntu-latest]
+        dart_sdk: ["3.3.0", stable]
     steps:
       - name: Checkout Code
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
       - name: Setup Dart
         uses: dart-lang/setup-dart@v1.3
         with:
           sdk: ${{ matrix.dart_sdk }}
-      - name: Cache Dart dependencies
-        uses: actions/cache@v3
+
+      - name: Cache Dart Dependencies
+        uses: actions/cache@v4
         with:
           path: ${{ env.PUB_CACHE_PATH }}
           key: ${{ runner.os }}-pub-cache-${{ matrix.dart_sdk }}
-          restore-keys: |
-            ${{ runner.os }}-pub-cache-
-      - name: Verify formatting
+          restore-keys: ${{ runner.os }}-pub-cache-
+
+      - name: Verify Formatting
+        # Disable formatting check on stable for now.
+        # Dart stable (3.7.2 as of 2025-03-14) has radically different formating
+        if: ${{ matrix.dart_sdk != 'stable' }}
         run: dart format --output=none --set-exit-if-changed .
 
-  dart_analyze:
-    name: Run Dart Analysis
-    runs-on: ${{ matrix.os }}
-    strategy:
-      matrix:
-        os: [windows-latest, ubuntu-latest, macos-latest]
-        dart_sdk: ['3.3.0', 'stable']
-    steps:
-      - name: Checkout Code
-        uses: actions/checkout@v3
-      - name: Setup Dart
-        uses: dart-lang/setup-dart@v1.3
-        with:
-          sdk: ${{ matrix.dart_sdk }}
-      - name: Cache Dart dependencies
-        uses: actions/cache@v3
-        with:
-          path: ${{ env.PUB_CACHE_PATH }}
-          key: ${{ runner.os }}-pub-cache-${{ matrix.dart_sdk }}
-          restore-keys: |
-            ${{ runner.os }}-pub-cache-
-      - name: Install dependencies
-        run: dart pub upgrade
-      - name: Analyze
-        run: dart analyze
+      - name: Check Dependency Bounds
+        if: ${{ matrix.dart_sdk != '3.3.0' }} # --tighten available from 3.5
+        run: >-
+          dart pub downgrade --tighten &&
+          dart pub upgrade --major-versions &&
+          git diff-files --quiet
```

Comment:
This helps us catch if dependencies are released in new major versions, and to see if our lower constraints are artificially low.